sensors/lego_sensor_class.c

```diff
@@ -326,8 +325,7 @@ static ssize_t commands_show(struct device *dev, struct device_attribute *attr,
 
 	for (i = 0; i < sensor->num_commands; i++)
 		count += sprintf(buf + count, "%s ", sensor->cmd_info[i].name);
-	if (count == 0)
-		return -EOPNOTSUPP;
+
 	buf[count - 1] = '\n';
```

Comment:
Have you tested this with a sensor with no commands? It seems to me that we would crash here if `count` == 0 because that would make the array index -1.

Maybe move the `if` statement before the `for` loop:

```
if (count == 0) {
        buf[0] = '\n';
        return 1;
}
```

Reply:
I thought the same thing, but the flow is identical to the case in [`tacho_motor_class.c`](https://github.com/ev3dev/lego-linux-drivers/blob/526851b7b7a2e90ff46febdc48a55735ef7d7bfd/motors/tacho_motor_class.c#L625-L643). Is this because there is never a case where `size` is zero there?

Reply:
Correct. In the tacho-motor class, we can assume that there will never be an empty list, but for sensors, there can be an empty list for `commands`.

Reply:
No worries, I fix that tonight (I only just noticed your replies - I have been spammed by our HPC queue telling me of hundreds of node failures and everything else just got pushed down).